analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyNullableValueAccessBase.cs

```diff
@@ -34,19 +34,24 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
             && reference.Property.Name == nameof(Nullable<int>.Value)
             && reference.Instance is { } instance
             && instance.Type.IsNullableValueType()
-            && context.HasConstraint(instance, ObjectConstraint.Null))
+            && context.HasConstraint(instance, ObjectConstraint.Null)
+            && NotNullFlowState(reference.Instance))
         {
             ReportIssue(instance, instance.Syntax.ToString());
         }
         else if (operationInstance.Kind == OperationKindEx.Conversion
             && operationInstance.ToConversion() is var conversion
             && conversion.Operand.Type.IsNullableValueType()
             && conversion.Type.IsNonNullableValueType()
-            && context.HasConstraint(conversion.Operand, ObjectConstraint.Null))
+            && context.HasConstraint(conversion.Operand, ObjectConstraint.Null)
+            && NotNullFlowState(conversion.Operand))
         {
             ReportIssue(conversion.Operand, conversion.Operand.Syntax.ToString());
         }
 
         return context.State;
+
+        bool NotNullFlowState(IOperation reference) =>
```

Comment:
The name seems wrong to me from looking at the implementation.
Implementation says (equivalent evolution on each line)
```
FlowState != NotNull
FlowState Not NotNull
Not NotNull FlowState
```
while the method is named
```
NotNull FlowState
```
so it's the opposite.

It's should be `NotNotNullFlowState` and that's bad. 

I don't have better idea for a good name. So we can avoid it like this:
```
        &&  FlowState(reference.Instance) != NullableFlowState.NotNull
// ...
        NullableFlowState FlowState(IOperation reference) =>
            SemanticModel.GetTypeInfo(reference.Syntax).Nullability().FlowState;
```